### PR TITLE
Create GitHub actions

### PR DESCRIPTION
**Pull request by connorIW**
Also remove redundant circleci actions.

### CODE REVIEW

**Review by mark-r-m — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        id: get-publish-version
+        run: |
+          LATEST_VERSION=$(curl --silent "https://api.github.com/repos/${{ github.repository }}/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+          [  -z "$LATEST_VERSION" ] && LATEST_VERSION="0.0.0"
+          VERSION=$(echo $LATEST_VERSION | awk -F. '{$NF = $NF + 1;} 1' | sed 's/ /./g')
+          echo "::set-output name=publish-version::${VERSION}"
+
+  publish-github-release:
+    runs-on: ubuntu-latest
+    needs: [get-publish-version, publish-docker]
```
Comment (dan-hill2802) [inline .github/workflows/main.yml:28]:
```suggestion
    needs: [get-publish-version, build-app]
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: build and test
+        run: make integration-all
```
Comment (dan-hill2802) [inline .github/workflows/main.yml:33]: this would make more sense in the `build-app` job

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Create GitHub Release
```
Comment (dan-hill2802) [inline .github/workflows/main.yml:34]: this step needs the jar specifying as an asset to upload

File: Makefile [hunk 4]
```diff
@@ -27,6 +27,10 @@ git-hooks: ## Set up hooks in .git/hooks
 
 local-build: ## Build Kafka2Hbase with gradle
 	gradle :unit build -x test
+    mkdir -p artifacts
+    cp -v README.md artifacts
+    cp -v build/distributions/*.tar artifacts/
+    gzip artifacts/*.tar
```
Comment (dan-hill2802) [inline Makefile:33]: would these make more sense in the `local-dist` command instead?

**Review by mark-r-m — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (connorIW) [reply Makefile]: I've removed this as reconciler application doesn't build like this.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  publish-github-release:
+    runs-on: ubuntu-latest
+    needs: [get-publish-version, build-app]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: Download built app
+        uses: actions/download-artifact@v2
+        with:
+          name: build-app
```
Comment (dan-hill2802) [inline .github/workflows/main.yml:38]:
```suggestion
          name: built-app
```

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**